packages/components/date-picker/src/date-picker-com/basic-year-table.vue

```diff
@@ -59,54 +67,177 @@ const startYear = computed(() => {
     : false
 
+  kls.today = today.year() === year
   kls.current =
     castArray(props.parsedValue).findIndex((d) => d!.year() === year) >= 0
 
-  kls.today = today.year() === year
+  if (cell.inRange) {
+    kls['in-range'] = true
 
+    if (cell.start) {
+      kls['start-date'] = true
+    }
+
+    if (cell.end) {
+      kls['end-date'] = true
+    }
+  }
   return kls
 }
 
-const isSelectedCell = (year: number) => {
-  return (
-    (year === startYear.value &&
-      props.date.year() < startYear.value &&
-      props.date.year() > startYear.value + 9) ||
-    castArray(props.date).findIndex((date) => date.year() === year) >= 0 ||
-    castArray(props.parsedValue).findIndex((date) => date?.year() === year) >= 0
-  )
+const isSelectedCell = (cell: YearCell) => {
+  const year = cell.text
+  return castArray(props.date).findIndex((date) => date.year() === year) >= 0
 }
 
 const handleYearTableClick = (event: MouseEvent | KeyboardEvent) => {
-  const clickTarget = event.target as HTMLDivElement
-  const target = clickTarget.closest('td')
-  if (target && target.textContent) {
-    if (hasClass(target, 'disabled')) return
-    const year = target.textContent || target.innerText
-    if (props.selectionMode === 'years') {
-      if (event.type === 'keydown') {
-        emit('pick', castArray(props.parsedValue), false)
-        return
-      }
-      const newValue = hasClass(target, 'current')
-        ? castArray(props.parsedValue).filter((d) => d?.year() !== Number(year))
-        : castArray(props.parsedValue).concat([dayjs(year)])
-      emit('pick', newValue)
+  const target = (event.target as HTMLElement)?.closest(
+    'td'
+  ) as HTMLTableCellElement
+  if (!target || !target.textContent || hasClass(target, 'disabled')) return
+
+  if (props.selectionMode === 'range') {
+    const column = target.cellIndex
+    const row = (target.parentNode as HTMLTableRowElement).rowIndex
+
+    const selectedYear = row * 4 + column + startYear.value
+    const newDate = dayjs().year(selectedYear)
+    if (!props.rangeState.selecting) {
+      emit('pick', { minDate: newDate, maxDate: null })
+      emit('select', true)
     } else {
-      emit('pick', Number(year))
+      if (props.minDate && newDate >= props.minDate) {
+        emit('pick', { minDate: props.minDate, maxDate: newDate })
+      } else {
+        emit('pick', { minDate: newDate, maxDate: props.minDate })
+      }
+      emit('select', false)
     }
+  } else if (props.selectionMode === 'years') {
+    const year = target.textContent || target.innerText
+    if (event.type === 'keydown') {
+      emit('pick', castArray(props.parsedValue), false)
+      return
+    }
+    const newValue = hasClass(target, 'current')
+      ? castArray(props.parsedValue).filter((d) => d?.year() !== Number(year))
+      : castArray(props.parsedValue).concat([dayjs(year)])
+    emit('pick', newValue)
+  } else {
+    const year = target.textContent || target.innerText
```

Comment:
The `selectedYear (L188)` and `year (L202/L212)` should be the same value. It is recommended to keep `selectedYear` and optimize the code.